### ISSUE TITLE
Framebuffer driver

### ISSUE THREAD

**Issue by alidaf** [ralph-irving/tcz-jivelite#3]
I have a framebuffer driver for a small OLED (256x64) working and would like to get this working with it if I can. I'd also like to be able to add a rotary encoder as an input device at some point to create a small standalone unit a bit like a boombox.

You mention framebuffers in the Wiki but there isn't any detail on how to use one other than modifying system variables. Does jivelite have it's own framebuffer driver or does it use X? Could you elucidate a little so I can work on it. The driver I have creates a framebuffer at /dev/fb1 but I believe that it is possible to mirror /dev/fb0 to /dev/fb1 and take advantage of hardware acceleration.

I will probably have to produce new skins, which is actually something I want to do anyway as I have a theme in mind to go with the case I have used. Where can I start for info?

Regards

**Comment by ralph-irving** [comment]
The tcz jivelite uses the SDL framebuffer driver to interface with the kernel fb? devices. See https://github.com/ralph-irving/squeezeplay/tree/master/src/SDL-1.2.15/src/video/fbcon   No X is needed to interface with the RPI frame buffer for the tcz jivelite build; only what's required by SDL fbcon.

There is very little documentation on creating jive lua skins.  I've only found the Developer Information section from the SD wiki http://wiki.slimdevices.com/index.php/SqueezeOS and looking at the lua code.

Hope that helps.

**Comment by alidaf** [comment]
Thanks for that. I've had a look at the SDL minefield and it seems totally unnavigable. I don't know how on Earth you managed it. I think though that it may be possible to just copy the linux framebuffer onto mine after I've forced the size down. I'll try it. I had intended to use a VFD from an old Popcorn player as it would look a lot like the Squeezebox III, but desoldering it was a bugger and now I just keep the smouldered remains as a shrine to Icarus. The closest and most affordable that I can get to reproduce the feel is an OLED but it's taken me months just to get this far.
